### PR TITLE
Remove FilterValuePlaceholder and add null handling for equals and add native filtercriterions

### PR DESCRIPTION
**Pull request by datazuul**
* remove FilterValuePlaceholder
* add null handling for equals and notequals filtering
* Add native filter criterions
* improve documentation

### CODE REVIEW

**Review by morpheus-87 — APPROVED** [review]

File: dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java [hunk 1]
```diff
@@ -7,9 +7,12 @@
  * Filter Criterion Container.A filter criterion is a composition of
  *
  * <ul>
- *   <li>a field name being target of filter operation
+ *   <li>an expression (e.g. field name) being target of filter operation
  *   <li>a filter operation (operator)
  *   <li>one ore more filter values to be used for filtering operation (operand(s))
+ *   <li>a flag to indicate if expression has to be nativeExpression or be used unchanged in
```
Comment (daforster) [inline dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java:13]:
```suggestion
 *   <li>a flag to indicate if expression has to be mapped or be used unchanged in
```

File: dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java [hunk 2]
```diff
@@ -7,9 +7,12 @@
...
+ *       underlying backend system (e.g. mapping Java member names to database column names; no
```
Comment (daforster) [inline dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java:14]:
```suggestion
 *       underlying backend system (e.g. mapping Java member names to database column names vs. no
```
Just to point it out more precisely

File: dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java [hunk 3]
```diff
@@ -21,35 +24,41 @@
...
 
   /**
-   * Constructor for Filter Criteria.
+   * Constructor for a filter criterion.
    *
-   * @param fieldName target field the criteria should be used for
-   * @param operation operation of criteria
-   * @param value operand of criteria
+   * @param expression expression the criterion should be used for
+   * @param nativeExpression a flag to indicate if expression has to be nativeExpression or should
```
Comment (daforster) [inline dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java:39]:
```suggestion
   * @param nativeExpression a flag to indicate if expression has to be mapped or should
```
I guess it was changed by the IDE when renaming the field

File: dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java [hunk 4]
```diff
@@ -109,17 +182,33 @@ public Object getValue() {
...
+  public void setExpression(String expression) {
+    this.expression = expression;
     validate();
   }
 
+  /**
+   * @param expression criterion expression
+   * @deprecated use setExpression(String expression) instead
+   */
+  public void setFieldName(String expression) {
```
Comment (daforster) [inline dc-model/src/main/java/de/digitalcollections/model/filter/FilterCriterion.java:198]:
```suggestion
  @Deprecated
    public void setFieldName(String expression) {
```

**Review by daforster — CHANGES_REQUESTED** [review]

**Review by clorenz — APPROVED** [review]

**Review by daforster — APPROVED** [review]

**Pull request merged**